posts/change-data-capture-cloudsql-postgres-with-apache-airflow-1.md

```diff
@@ -0,0 +1,1002 @@
+
+    ```bash
+    https://www.googleapis.com/auth/cloud-platform
+    ```
+
+3. In the `Keyfile JSON` field, paste the Keyfile JSON you copied from Step 2. 
+
+    ![](../assets/cdc-cloudsql-1/cdc_cloudsql_airflow_2021-01-13_at_4.25.24_PM.png)
+
+Keep the `Conn Id` handy. You will use that in your Airflow DAG to reference this connection. When you're ready, save this connection to finalize your access management. 
+
+## Step 6: Configure Pooling
+
+The next configuration you'll set is a pool. From the [Airflow docs](https://airflow.apache.org/docs/apache-airflow/stable/concepts.html?highlight=pool#pools), `Airflow pools can be used to limit the execution parallelism on arbitrary sets of tasks`, and therefore, using them will enable you to create other parallel jobs while still requiring the export API tasks to be guaranteed to be run in serial. 
+
+To create this is simple. Just navigate in the top bar to `Admin -> Pools`, then hit the `+` sign to add a new record. Choose a pool name and description to your liking, but choose the number of slots as `1`.
+
+![](../assets/cdc-cloudsql-1/cdc_cloudsql_airflow_2021-01-14_at_11.05.02_AM.png)
+
+Will add the name of the pool to your task definition below and that's it.
+
+## Step 7: Build Your Pipeline in Airflow
+
+  > *If you are familiar with the foundational concepts of Airlfow, you can skip to the next subsection for the `DAG Workflow`*
+
+Airflow pipeline design rests on the foundational structures of `Hooks` and `Operators`. Hooks facilitate connections and repository-type actions, whereas Operators use hooks in combinations with domain-specific logic and error handling in order to accomplish a task. Airflow comes pre-packaged with many of these hooks and operators, all built to work with services we use regularly. For example, `BashOperator` allows you to run a bash command directly inline. A simple task using a this operator might look like:
+
+```python
+hello_world = BashOperator(
+    task_id='get_schema',
+    bash_command='echo "hello, world"'
+)
+```
+
+Getting more advanced, a `PostgresOperator` executes SQL code in a specific PostgreSQL database using a `PostgresHook` to establish the connection, while templating your command and implementing a `run` function to execute it. A task involving this operator might look like:
+
+```python
+insert_row = PostgresOperator(
+    task_id='insert_row',
+	  postgres_conn_id='default_conn_id',
+    sql='INSERT INTO table_1 VALUES (%s, %s, %s)',
+    parameters=(1, 'a very important string', datetime.now())
+)
+```
+
+### DAG Workflow
+
+Because of the great flexibility of Airflow, you can also use Hooks to create custom Operators, just as well as you can create custom hooks. In this post, we will create a custom `CloudSqlCsvExportOperator` using an existing `CloudSqlHook`, a hook designed to facilitate several types of operations on CloudSQL instances. First, we will use the `BashOperator` to simulate our workflow and check dependencies.
```

Comment:
Same way, standardize on capitalizing Hooks and Operators.

Reply:
going with lowercase, makes more sense